ci(actions): update scripts/workflows to support lifecycle label changes (#13256)
Esri/calcite-components: CONTRIBUTING.md

### Lifecycle

> There are five issue lifecycle labels:

Reviewer: Suggestion: we could avoid having to update this in the future if the label count changes by not calling out the total amount of labels. I'll defer to you since these labels don't change often.

Reviewer 2: +1 on keeping this more generalized for future maintenance.

Author: Sounds good to me! WDYT about merging the line below the list with the intro?

```suggestion
Lifecycle labels are used to communicate the state of an issue, and so an issue can only have one of them at any time. Please make sure to keep these up to date.
```

Reviewer 2: nitpick on the wording, but dig this approach! 💪🏻 

> Lifecycle labels are used to communicate the state of an issue. Each issue can only be assigned one lifecycle label at any time. Please make sure to keep the lifecycle label up to date.

> - `1 - in design`: Issue that are actively undergoing design work.

Reviewer 2: ```suggestion
- `1 - in design`: Issues that are actively undergoing design expertise.
```

- `2 - ready for dev`: Issues that are fully designed, have acceptance criteria defined, and are ready for a developer to pick up.

> - `3 - in development`: Issues that are actively being worked on.

Reviewer 2: Do we want to add more context to development work with the new addition of the `in design` label for more specificity?

```suggestion
- `3 - in development`: Issues that are actively under development.
```

- `4 - installed`: Issues that have been merged to the `dev` branch and/or are ready for QA/QC.
- `5 - verified`: Issues that have been tested, confirmed as mitigated, and are ready to close.

> An issue can only have one of the lifecycle labels at any time. Please make sure to keep these up to date.

Reviewer 2: Even though this is a bit out of scope for this PR, wondering if we should call out this out before listing out the lifecycle labels?

Author: Ha, didn't even see this before making [this suggestion](https://github.com/Esri/calcite-design-system/pull/13256/files#r2491689549). Agreed 👍

### Issues that cannot be worked on

There are four labels that mean an issue is not ready for development:

> - `1 - in design`: Issues that are going through a design phase. Once design completes their work or review, the `2 - ready for dev` label will be added to the issue, which means a developer can pick up the issue.

Reviewer 2: nitpick, but wondering if we want to call it a phase?

```suggestion
- `1 - in design`: Issues that are going through a design consultation. Once designers complete the effort, the `2 - ready for dev` label will be added to the issue, which means a developer can pick up the issue.
```
